src/app/fyle/spender-onboarding/spender-onboarding.page.ts

```
@@ -121,17 +125,18 @@
 
   markStepAsComplete(): void {
     if (this.currentStep === OnboardingStep.CONNECT_CARD) {
-      this.spenderOnboardingService.markConnectCardsStepAsComplete().subscribe(() => {
-        this.currentStep = OnboardingStep.OPT_IN;
-      });
-    }
-    if (this.currentStep === OnboardingStep.OPT_IN) {
+      this.spenderOnboardingService
+        .markConnectCardsStepAsComplete()
+        .pipe(map(() => (this.currentStep = OnboardingStep.OPT_IN)))
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)